mixer/pkg/runtime/dispatcher/session.go

```diff
@@ -194,8 +202,11 @@ func (s *session) dispatch() error {
 	s.waitForDispatched()
 
 	if s.variety == tpb.TEMPLATE_VARIETY_QUOTA && !foundQuota {
-		log.Errorf("Requested quota '%s' is invalid", s.quotaArgs.Quota)
-		s.err = multierror.Append(s.err, fmt.Errorf("requested quota '%s' is invalid", s.quotaArgs.Quota))
+		// If quota is not found it is very likely that quotaSpec / quotaSpecBinding was applied first
+		// We still err on the side of allowing access, but warn about the fact that quota was not found.
```

Comment:
I don't think we should have this behavior. Incorrect config will lead to quotas not being enforced. Mixer should have systematic fail-close semantics and leave it up to the proxy (the final customer) to decide whether fail-open is appropriate.

Reply:
I am updating  quota api to support these semantics. We are unable to provide this info to the client at this time.

It’s much easier to change impl right now.

Returning an error here causes needless intermittent denials simply because the 5 config elements needed for quota were applied in an incorrect order.

I think It is better for quotas to be biased towards allowing access.

Reply:
If you're referring to config consistency issues, then generally in istio project we've been ok with manual staging. E.g. push quota/quotaspecbinding config first, wait a couple of minutes, then apply mixer rules. That's how virtual service + destination rule staging works.

Reply:
Does it cause outage if virtual service and destination rules are applied out of order?  If there is absolutely no other way then asking for specific staging is ok.

However if we operate under the assumption of pre validated config that is validated offline or by galley, then the most likely cause of inconsistency at runtime is staging, which can be trivially addressed.